contracts/AelinPool.sol

```diff
@@ -97,18 +105,31 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         aelinDealLogicAddress = _aelinDealLogicAddress;
         aelinRewardsAddress = _aelinRewardsAddress;
 
-        if (_poolData.allowListAddresses.length > 0 || _poolData.allowListAmounts.length > 0) {
+        address[] memory tmpAllowListAddresses = _poolData.allowListAddresses;
+        uint256[] memory tmpAllowListAmounts = _poolData.allowListAmounts;
+
+        if (tmpAllowListAddresses.length > 0 || tmpAllowListAmounts.length > 0) {
             require(
-                _poolData.allowListAddresses.length == _poolData.allowListAmounts.length,
+                tmpAllowListAddresses.length == tmpAllowListAmounts.length,
                 "allowListAddresses and allowListAmounts arrays should have the same length"
             );
-            for (uint256 i = 0; i < _poolData.allowListAddresses.length; i++) {
-                allowList[_poolData.allowListAddresses[i]] = _poolData.allowListAmounts[i];
-                emit AllowlistAddress(_poolData.allowListAddresses[i], _poolData.allowListAmounts[i]);
+            for (uint256 i = 0; i < tmpAllowListAddresses.length; i++) {
+                allowList[tmpAllowListAddresses[i]] = tmpAllowListAmounts[i];
+                emit AllowlistAddress(tmpAllowListAddresses[i], tmpAllowListAmounts[i]);
             }
             hasAllowList = true;
         }
 
+        NftData[] memory tmpNftData = _poolData.nftData;
+
+        if(tmpNftData.length > 0) {
+            require(tmpNftData.length <= 5, "max 5 collections allowed");
+            for(uint256 i = 0; i < tmpNftData.length; i++) {
```

Comment:
we need to emit an event for every contract we add along with the rules for that NFT collection in the pool. We will use this event in the subgraph, which the client will read to tell a user how much they can invest and which pools are available for a given collection.

Reply:
also we need to make sure that if there is a 721 collection passed in here we do not also pass in an 1155 collection as well since you can only have one type per pool. we can lump cryptopunks in with 721. We can hardcode the address in the contracts and do an if statement to capture that address when it is passed in alongside 721s or by itself

Reply:
will add events to it. and can we keep 721 and 1155 in the same pool and use seperate functions for both of them. it'll add more flexibility to the sponsor.